R/rlang.R

```diff
@@ -5,18 +5,24 @@ NULL
 is_same_body <- NULL
 
 
+defer_if_unloaded <- function(pkg, fn, action = NULL) {
```

Comment:
I was expecting something more like this:

```R
on_pkg_load <- function(pkg, code) {
  if (isNamespaceLoaded(pkg)) {
    code
  } else {
    f <- function() code
    setHook(packageEvent(pkg, "onLoad"), f)
  }
}
```

I find that a bit easier to reason about, rather than a hook to re-call the current function plus a non-local return.